core/types/transaction.go

```diff
@@ -29,21 +29,21 @@ import (
 	"github.com/ethereum/go-ethereum/rlp"
 )
 
-//go:generate gencodec -type txdata -field-override txdataMarshaling -out gen_tx_json.go
+//go:generate gencodec -type Txdata -field-override TxdataMarshaling -out gen_tx_json.go
 
 var (
 	ErrInvalidSig = errors.New("invalid transaction v, r, s values")
 )
 
 type Transaction struct {
-	data txdata
+	data Txdata
 	// caches
 	hash atomic.Value
 	size atomic.Value
 	from atomic.Value
 }
 
-type txdata struct {
+type Txdata struct {
```

Comment:
How come `Txdata` needs to be made public?  I don't see it being used outside of the `types` packages in this PR.

Reply:
@yorhodes can provide more context, but while inspecting the transaction on the client, he ended up requiring this